tests/test_series.py

```diff
@@ -3069,3 +3070,103 @@ def first_arg_not_series(argument_1: int, ser: pd.Series) -> pd.Series:
             ),
             1,
         )
+
+
+def test_diff() -> None:
+    s = pd.Series([1, 1, 2, 3, 5, 8])
+    # int -> float
+    check(assert_type(s.diff(), "pd.Series[float]"), pd.Series, float)
+    # unint -> float
+    check(assert_type(s.astype(np.uint32).diff(), "pd.Series[float]"), pd.Series, float)
+    # float -> float
+    check(assert_type(s.astype(float).diff(), "pd.Series[float]"), pd.Series, float)
+    # datetime.date -> timeDelta
+    check(
+        assert_type(
+            pd.Series(
+                [datetime.datetime.now().date(), datetime.datetime.now().date()]
+            ).diff(),
+            "TimedeltaSeries",
+        ),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # timestamp -> timedelta
+    times = pd.Series([pd.Timestamp(0), pd.Timestamp(1)])
+    check(
+        assert_type(times.diff(), "TimedeltaSeries"),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # timedelta -> timedelta64
+    check(
+        assert_type(
+            pd.Series([pd.Timedelta(0), pd.Timedelta(1)]).diff(), "TimedeltaSeries"
+        ),
+        pd.Series,
+        pd.Timedelta,
+        index_to_check_for_type=-1,
+    )
+    # period -> object
+    with pytest_warns_bounded(
+        RuntimeWarning, "overflow encountered in scalar multiply"
+    ):
```

Comment:
this shouldn't be necessary

Reply:
it is for pytest to run successfully : when running diff of Series[Period] a RuntimeWarning is raised. I guess it is close to be a pandas core bug (or maybe numpy)

Reply:
I think that's a pandas bug.  So can you change your test to:
```python
pd.Series(pd.period_range(start='2017-01-01', end='2017-02-01', freq='D')).d
iff()
```
That won't show the bug

Reply:
still have the same issue with `pd.period_range`

``` python
___________________________________________________________________________________________________ test_diff ____________________________________________________________________________________________________ 

    def test_diff() -> None:
        s = pd.Series([1, 1, 2, 3, 5, 8])
        # int -> float
        check(assert_type(s.diff(), "pd.Series[float]"), pd.Series, float)
        # unint -> float
        check(assert_type(s.astype(np.uint32).diff(), "pd.Series[float]"), pd.Series, float)
        # float -> float
        check(assert_type(s.astype(float).diff(), "pd.Series[float]"), pd.Series, float)
        # datetime.date -> timeDelta
        check(
            assert_type(
                pd.Series(
                    [datetime.datetime.now().date(), datetime.datetime.now().date()]
                ).diff(),
                "TimedeltaSeries",
            ),
            pd.Series,
            pd.Timedelta,
            index_to_check_for_type=-1,
        )
        # timestamp -> timedelta
        times = pd.Series([pd.Timestamp(0), pd.Timestamp(1)])
        check(
            assert_type(times.diff(), "TimedeltaSeries"),
            pd.Series,
            pd.Timedelta,
            index_to_check_for_type=-1,
        )
        # timedelta -> timedelta64
        check(
            assert_type(
                pd.Series([pd.Timedelta(0), pd.Timedelta(1)]).diff(), "TimedeltaSeries"
            ),
            pd.Series,
            pd.Timedelta,
            index_to_check_for_type=-1,
        )
        # period -> object
        check(
            assert_type(
                pd.Series(
                    pd.period_range(start="2017-01-01", end="2017-02-01", freq="D")
>               ).diff(),
                "pd.Series[type[object]]",
            ),
            pd.Series,
            object,
        )

tests\test_series.py:3117:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\series.py:3111: in diff
    result = algorithms.diff(self._values, periods)
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\algorithms.py:1379: in diff
    return op(arr, arr.shift(n))
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\ops\common.py:76: in new_method
    return method(self, other)
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\arrays\datetimelike.py:1460: in __sub__
    result = self._sub_periodlike(other)
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\arrays\datetimelike.py:1304: in _sub_periodlike
    new_data = np.array([self.freq.base * x for x in new_i8_data])
C:\Users\laurent.mutricy\AppData\Local\miniconda3\envs\pandas-stubs\Lib\site-packages\pandas\core\arrays\datetimelike.py:1304: in <listcomp>
    new_data = np.array([self.freq.base * x for x in new_i8_data])
offsets.pyx:1033: in pandas._libs.tslibs.offsets.Tick.__mul__
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _  

>   ???
E   RuntimeWarning: overflow encountered in scalar multiply

offsets.pyx:487: RuntimeWarning
```

Reply:
It's Windows only when the frequency is "D".  So if we use "h", we don't get the warning.